velox/exec/SortWindowBuild.cpp

```diff
@@ -334,16 +358,24 @@ void SortWindowBuild::loadNextPartitionFromSpill() {
     sortedRows_.push_back(newRow);
     next->pop();
   }
-}
 
-std::shared_ptr<WindowPartition> SortWindowBuild::nextPartition() {
-  if (merge_ != nullptr) {
-    VELOX_CHECK(!sortedRows_.empty(), "No window partitions available");
-    auto partition = folly::Range(sortedRows_.data(), sortedRows_.size());
-    return std::make_shared<WindowPartition>(
-        data_.get(), partition, inversedInputChannels_, sortKeyInfo_);
+  // No more partition batches. All data is consumed.
+  if (sortedRows_.empty()) {
+    partitionStartRows_.clear();
+    numSpillReadBatches_--;
+
+    auto lockedOpStats = opStats_->wlock();
+    auto pair = lockedOpStats->runtimeStats.try_emplace(
+        Window::kSpillWindowReadBatchNums, RuntimeMetric(numSpillReadBatches_));
+    if (!pair.second) {
```

Comment:
Why do we need this conditional logic? Aren't we setting these stats exactly once?